### PR TITLE
Make more use of async/await

### LINKED ISSUE #254: Fully transition to async/await, tokio 0.2 or async-std
#253 already did a big step towards making more use of async/await. However, we cannot fully complete this yet because `shiplift` still depends on `tokio` 0.1 internally.

### PR DESCRIPTION
**Pull request by thomaseizinger**
This PR makes better use of async/await inside CCA.

By relying on tokio 0.2 and tokio-compat, we can remove the dependency on async-std and signal-hook.

With this, we also now gain the possibility of cancelling the startup _at any time_ and not just between the individual services.

Fixes #254. (When #254 was created, `tokio-compat` did not yet exist, hence it was blocked by shiplift transitioning to tokio 0.2)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Upgrade all examples and new project to `comit-sdk@0.8.0`.
 - To create new project, `new` is not needed in arguments anymore: `yarn create comit-app foo`.
 - New binary `comit-scripts` to access other create-comit-app features. E.g.: `comit-scripts start-env`.
+- Pressing CTRL+C while `start-env` is running is now handled immediately and not just after the current service is fully started.
```
Comment (D4nte) [inline CHANGELOG.md]: Mention that we do not clean after a panic anymore.

File: Cargo.toml [hunk 2]
```diff
@@ -25,12 +24,12 @@ secp256k1 = { version = "0.12", features = ["rand"] }
...
 serde_json = "1"
 shiplift = { version = "0.6", default-features = false }
-signal-hook = { version = "0.1", features = ["tokio-support"] }
 structopt = "0.3"
-tar = "0.4.26"
+tar = "0.4"
 tempfile = "3.1.0"
 thiserror = "1"
-tokio = "0.1"
+tokio = { version = "0.2", features = ["full"] }
```
Comment (D4nte) [inline Cargo.toml]: I am guessing it'd save some compile time if you only select the necessary features.

File: build.rs [hunk 3]
```diff
@@ -1,16 +1,15 @@
 use anyhow::Context;
-use flate2::{write::GzEncoder, Compression};
 use ignore::WalkBuilder;
 use std::{env, fs::File, path::Path};
 
 fn main() -> anyhow::Result<()> {
     let out_dir = env::var("OUT_DIR").context("unable to read OUT_DIR variable")?;
     let out_dir = Path::new(&out_dir);
-    let archive = out_dir.join("new_project.tar.gz");
+    let archive = out_dir.join("new_project.tar");
```
Comment (D4nte) [inline build.rs]: Why?

File: src/env/mod.rs [hunk 4]
```diff
@@ -1,82 +1,60 @@
...
-    .expect("Clean up failed");
-    println!("Clean up done!");
-}
-
-pub fn start() {
-    let mut runtime = Runtime::new().expect("Could not get runtime");
-
-    if crate::temp_fs::dir_exist() {
+pub async fn start() {
+    if crate::temp_fs::dir_exist().await {
```
Comment (D4nte) [inline src/env/mod.rs]: noice.

File: build.rs (hunk 3 above)
Comment (thomaseizinger) [reply build.rs]: https://github.com/comit-network/create-comit-app/pull/305/commits/258e679503b9eda9277b55b1f56d752ce022cdbf

**Comment by thomaseizinger** [comment]
Happy new year to you too @bors! :tada:

**Pull request merged**